### ISSUE TITLE
Delete files without confirmation

### ISSUE THREAD

**Issue by ksevelyar** [jarun/nnn#651]
Hi, is it possible to delete files/dirs without confirmation with `x`?

**Comment by 0xACE** [comment]
Confirmation of what? The sel/cur option, or the `rm -rf`?

You would have to patch the source code anyhow. It has been discussed at least twice before, you can search amongst the old issues if you are interested or look up `SEL_RM` in https://github.com/jarun/nnn/blob/251a9a2e9c7172682a14da47a1270e4ea8bc7ff7/src/nnn.c#L6117

**Comment by ksevelyar** [comment]
Thank you for the quick response! 
I'll pick another fm then.

**Comment by jarun** [comment]
Relevant notes here: https://github.com/jarun/nnn/issues/626#issuecomment-637903576

**Comment by jarun** [comment]
I forgot we have the auto-lock bot in place now. :)

**Comment by jarun** [comment]
@0xACE now that we have 3 issues where users are asking to prefer selection when it exists, do you think it makes sense to have a program option?

**Comment by leovilok** [comment]
@ksevelyar you can also easily create a plugin like
~~~sh
#!/bin/sh
xargs -0 rm -rf < ${XDG_CONFIG_HOME:-$HOME/.config}/nnn/.selection
~~~
to remove selection or
~~~sh
#!/bin/sh
rm -rf "$1"
~~~
to remove current, give it plugin key `x`, and call it with a single <kbd>Alt</kbd>+<kbd>x</kbd>. (Use it at your own risks, as this doesn't asks for confirmation, of course)

**Comment by jarun** [comment]
I think users are looking for a GUI-like behaviour - when you select multiple files (visually, say) the next operation is on the selection, no questions asked. So on second thoughts, I preferred to open a discussion. We have multiple requests already.

@leovilok I can't find a suitable program option (switch letter) right away, can we help me with that?

**Comment by jarun** [comment]
There's another aspect... if we do this in `nnn`, it also makes sense to follow suit in plugins. In that case we need to have a config env variable (so plugins can read that too) rather than a program option.

**Comment by leovilok** [comment]
@jarun `-u` for "*use selection*", `-D` for "*don't ask about using selection*"?

**Comment by jarun** [comment]
Probably I'll leave out the plugins, because there's no easy way for `nnn` to know which one uses and which one doesn't use selection (and clear the selection after invoking it).

`-u` for _use selection always_` is nice! Thanks!

**Comment by leovilok** [comment]
oh, you're right, some plugins have similar UI ... but plugins can be edited

**Comment by jarun** [comment]
> but plugins can be edited

Yes, that's what I'm banking on. Asking the question doesn't lead to any data loss so we are good.

**Comment by jarun** [comment]
@ksevelyar I have implemented the enhancement at commit 4b4bf6cca55e7f6710113e36c5ff8f776ca045a2 with option `-u`. Please confirm it works as you expect.

**Comment by jarun** [comment]
Locking this thread as the patch is confirmed to work an another thread.
